### PR TITLE
added -targets flag to list all available build targets

### PR DESCRIPTION
**Pull request by cyberkm**
Show list of available build targets when -targets flag added

### CODE REVIEW

File: Code/Tools/FBuild/FBuildApp/Main.cpp [hunk 1]
```diff
@@ -604,3 +620,20 @@ int WrapperIntermediateProcess( const AString & args, const FBuildOptions & opti
 }
 
 //------------------------------------------------------------------------------
+// DisplayTargetsList
```
Comment (ffulin) [inline Code/Tools/FBuild/FBuildApp/Main.cpp]: Each entry in the hash table is a linked list. You'll need to walk the chain (m_Next) or you won't find all the nodes when there are collisions.
You'll likely need to move this function into the NodeGraph in order to be able to access the m_Next member of Node.

File: Code/Tools/FBuild/FBuildCore/Graph/NodeGraph.cpp [hunk 2]
```diff
@@ -523,6 +523,13 @@ Node * NodeGraph::GetNodeByIndex( uint32_t index ) const
 	return n;
 }
```
Comment (ffulin) [inline Code/Tools/FBuild/FBuildCore/Graph/NodeGraph.cpp]: If you move the DisplayTargetList function into NodeGraph, you won't need to expose this.

**Comment by ffulin** [comment]
This change looks good. I'll integrate it soon.

**Comment by ffulin** [comment]
I've integrated this change into the P4 mainline. It will be in the next release (v0.89).

I made small change during integration: The cmdline option is now "-showtargets" instead of "-targets".

**Comment by cyberkm** [comment]
Fine by me, thanx :)

**Pull request merged**